src/test/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCaseFlagsTest.java

```diff
@@ -70,15 +80,45 @@ void shouldSuccessfullyUpdateCaseFlags() {
         systemMigrateCaseFlags.setMigrationFlagEnabled(true);
 
         //When
-        AboutToStartOrSubmitResponse<CaseData, State> response =
+        final AboutToStartOrSubmitResponse<CaseData, State> response =
             systemMigrateCaseFlags.aboutToSubmit(updatedCaseDetails, beforeDetails);
 
         //Then
-        assertThat(response.getData()).isNotNull();
-        assertThat(response.getData().getCaseFlags()).isNotNull();
-        assertThat(response.getData().getSubjectFlags()).isNotNull();
-        assertThat(response.getData().getRepresentativeFlags()).isNotNull();
-        assertThat(response.getData().getApplicantFlags()).isNotNull();
+        assertNotNull(response.getData());
+        assertNotNull(response.getData().getCaseFlags());
+        assertNotNull(response.getData().getSubjectFlags());
+        assertNotNull(response.getData().getRepresentativeFlags());
+        assertNotNull(response.getData().getApplicantFlags());
     }
 
+    @Test
+    void shouldSuccessfullyUpdateCaseFlagsWithNullValues() {
```

Comment:
Consider adding a test for updating CaseFlags when `setMigrationFlagEnabled` is `false`